syft/frameworks/torch/tensors/interpreters/polynomial.py

```diff
@@ -1,257 +1,355 @@
+
+    The polynomial tensor provides flexibility to consider every non-linear function as piecewise linear function
+    and fit over different intervals.
+    
+    Args:
+        function:
+        precision:
+    """
+
+    def __init__(
+        self,
+        method="interpolation",
+        precision=10,
+        child=None,
+        owner=None,
+        id=None,
+        tags: set = None,
+        description: str = None,
+    ):
+        """
+        Args:
+            function[callable,Optional]: Function to applied to function approximation coefficients.
+                Used to encrypt coefficients.
+            precision[integer]: Precision of approximated values
+        """
+
+        super().__init__(owner=owner, id=id, tags=tags, description=description)
+
+        self.method = method
+        self.precision = precision
+        # Stores parameters of function approximations such as precision, degree, piecewise functions and base function
+        self.function_attr = {}
+        # Stores fitted function
+        self.func_approx = {}
+        self.child = child
+        self.id = None
+        self.encrypt_fn = {"exp": None, "sigmoid": None}
+
+        def default_functions():
+            """Initializes default function approximations exp, log, sigmoid and tanh"""
+
+            self.add_function(
+                "exp", lambda x: np.exp(x), degree=10, min_val=-10, max_val=10, steps=100
+            )
+            # self.add_function("log", 10,-10,10,10, lambda x: np.log(x))
+            self.add_function(
+                "sigmoid",
+                (lambda x: 1 / (1 + np.exp(-x))),
+                degree=10,
+                min_val=-10,
+                max_val=10,
+                steps=100,
+            )
+            self.add_function(
+                "tanh", lambda x: np.tanh(x), degree=10, min_val=-10, max_val=10, steps=100
+            )
+
+        default_functions()
+        self.apply_coefs()
+
+    def apply_coefs(self):
+        """The function sets the values of coefficients into interpolation/Taylor series based on 
+           the method set by the user.
+        """
+
+        if self.method == "taylor":
+
+            self.sigmoid_coeffs = [
+                (1 / 2),
+                (1 / 4),
+                0,
+                -(1 / 48),
+                0,
+                (1 / 480),
+                0,
+                -(17 / 80640),
+                0,
+                (31 / 1451520),
+            ]
+            self.exp_coeffs = [1, (1 / 2), (1 / 6), (1 / 24), (1 / 120), (1 / 840), (1 / 6720)]
+            self.tanh_coeffs = [(1), 0, (-1 / 3), 0, (2 / 15), 0, (-17 / 315)]
+
+        else:
+
+            self.sigmoid_coeffs = torch.tensor(self.func_approx["sigmoid"].coef)
+            self.exp_coeffs = torch.tensor(self.func_approx["exp"].coef)
+            self.tanh_coeffs = torch.tensor(self.func_approx["tanh"].coef)
+
+    def get_encrypt_function(self):
```

Comment:
Are you filling a dict with the coefficients for all the functions here?
If so couldn't you wait to use a function to create the appropriate tensor?

And also, why are `self.exp_coeffs`, etc. lists and not directly tensors

Reply:
Thanks for the suggestion, Jason. This is a refractor of the old PR. I realized its inefficient considering how only the required coefficients could be used when required.